OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java

```diff
@@ -482,11 +485,24 @@ private static OSEmailSubscriptionState getLastEmailSubscriptionState(Context co
    }
    // End EmailSubscriptionState
 
+   /**
+    * Get the current user data, notification and permissions state.
+    */
    @Nullable
    public static OSDeviceState getDeviceState() {
-      if (getPermissionSubscriptionState() == null)
+      // If applicable, check if the user provided privacy consent
+      if (shouldLogUserPrivacyConsentErrorMessageForMethodName("getDeviceState()"))
```

Comment:
Consent isn't needed since this is just a getter, it doesn't make any network requests